### PR TITLE
[EdgeTpu] Introduce EdgeTpuConfigSetting

### PR DESCRIPTION
**Pull request by imsw0529**
Related to issue #1621, draft #1639

This commit introduce EdgeTpuConfigSetting that has ConfigObject logic for EdgeTpu Backend

- Introduce class EdgeTpuConfigSetting extends ConfigSetting for EdgeTpu Backend logic
- Apply EdgeTpuConfigSetting to ConfigObject
- Add Test code for EdgeTpuConfigSetting

Co-author: SeungHyunH <vkflek1232@naver.com>

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

### CODE REVIEW

**Comment by imsw0529** [comment]
We have a question.
https://github.com/Samsung/ONE-vscode/pull/1639#issuecomment-1729055215
> ### About Backend detection
> We supposed that when there are edge-tpu-backend associated files(.edgetpucfg or model_edgetpu.tflie ans so on), How OneExplorer would treat it. There are two approach:
> 
> 1. **Do not show** edge-tpu-backend associated files **even though** there are that files in workspace when edge-tpu-backend is not installed.
> 2. **Show** edge-tpu-backend and **block** function **run cfg** if edge-tpu-backend is not installed.
> 
> We think that second approach is appropriate. What is your opinion?

**Review by SeungHyunH — APPROVED** [review]

File: src/OneExplorer/ConfigObject.ts [hunk 1]
```diff
@@ -140,6 +148,10 @@ export class ConfigObj {
     this.uri = uri;
     this.rawObj = rawObj;
     this.configType = CfgType.one;
+    const ext = path.extname(uri.fsPath);
+    if (ext === EdgeTpuConfigSetting.ext) {
+      this.configType = CfgType.edgeTpu;
+    }
```
Comment (dayo09) [inline src/OneExplorer/ConfigObject.ts:154]: How about tidying these by setting configType with similar conditional statement?

Comment (SeungHyunH) [reply src/OneExplorer/ConfigObject.ts:154]:
```
    let configSetting: ConfigSetting;
    this.configType = CfgType.one;
    const ext = path.extname(uri.fsPath);
    if (ext === EdgeTpuConfigSetting.ext) {
      this.configType = CfgType.edgeTpu;
    }
    switch (this.configType) {
      case CfgType.edgeTpu:
        configSetting = new EdgeTpuConfigSetting();
        break;
      case CfgType.one:
      default:
        configSetting = new OneConfigSetting();
    }
    this.uri = uri;
    this.rawObj = rawObj;
```
There seems to be an issue where you can set CfgType to edgetpu after creating OneConfigSetting. What do you think about this approach?

Comment (dayo09) [inline src/OneExplorer/ConfigObject.ts:154]:
```suggestion
    const ext = path.extname(uri.fsPath);
    switch(ext) {
      case EdgeTpuConfigSetting.ext:
        this.configType = CfgType.edgeTpu;
        break;
      case OneConfigSetting.ext:
      default:
        this.configType = CfgType.one;
    }
```

I meant, like this..?

Comment (imsw0529) [reply src/OneExplorer/ConfigObject.ts:154]: Thank you. I understand it. I will apply this :-D

**Review by dayo09 — APPROVED** [review]
LGTM, Thanks!! :-D

**Pull request merged**